java/ql/test/library-tests/frameworks/apache-commons-lang3/Test.java

```diff
@@ -0,0 +1,276 @@
+import org.apache.commons.lang3.StringUtils;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+class Test {
+    String taint() { return "tainted"; }
+
+    void sink(Object o) {}
+
+    void test() throws Exception {
+
+        // All these calls should convey taint to `sink` except as noted.
+        sink(StringUtils.abbreviate(taint(), 0));
```

Comment:
These sort of data flow tests quickly become unwieldy - perhaps it's time to slowly start using the inline test expectation library?